sympy/solvers/ode/single.py

```diff
@@ -359,6 +358,150 @@ def inverse(self):
+                  x0               y0
+
+    Where the first partials of `P` and `Q` exist and are continuous in a
+    simply connected region.
+
+    A note: SymPy currently has no way to represent inert substitution on an
+    expression, so the hint ``1st_exact_Integral`` will return an integral
+    with `dy`.  This is supposed to represent the function that you are
+    solving for.
+
+    Examples
+    ========
+    >>> from sympy import Symbol,Function
+    >>> from sympy import E,dsolve,Eq,log,sin,cos
+    >>> x=Symbol('x')
+    >>> f=Function('f')
+    >>> eq=Eq(0,(sin(x)*sin(f(x))-x*E**f(x))*f(x).diff(x)-(E**f(x)+cos(x)*cos(f(x))))
+    >>> dsolve(eq,hint='1st_exact')
+    Eq(x*exp(f(x)) + sin(x)*cos(f(x)), C1)
+    >>> dsolve(cos(f(x)) - (x*sin(f(x)) - f(x)**2)*f(x).diff(x),f(x), hint='1st_exact')
+    Eq(x*cos(f(x)) + f(x)**3/3, C1)
+
+    References
+    ==========
+
+    - "Differential Equations With Applications And Historical Notes",
+       Second Edition, George.F.Simmons
+    - https://en.wikipedia.org/wiki/Exact_differential_equation
+
+    # indirect doctest
+
+    """
+    hint="1st_exact"
+    has_integral=True
+    y = Dummy('y') # This is the only way to pass dummy y to _handle_Integral
+
+    def _wilds(self, f, x, order):
+        P = Wild('P')
+        Q = Wild('Q', exclude=[f(x).diff(x)])
+        return P, Q
+
+    def _equation(self, fx, x, order):
+        P, Q = self.wilds()
+        return P + Q*fx.diff(x)
+
+    def _verify(self, fx) -> bool:
+        P, Q = self.wilds()
+        eq = self.ode_problem.eq
+        x = self.ode_problem.sym
+        y = self.y
+
+        #This checks if Derivative present in wild P
+        if fx.diff(x) in self._wilds_match[P].atoms(Derivative):
+            eq = collect(eq,fx.diff(x))
+            pattern = self._equation(fx, x, 1)
+            self._wilds_match = eq.match(pattern)
+
+        m, n = self.wilds_match()
+        try:
+            if simplify(m) != 0:
+                m = m.subs(fx,y)
+                n = n.subs(fx,y)
+                numerator= simplify(m.diff(y)-n.diff(x))
+                # The following few conditions try to convert a non-exact
+                # differential equation into an exact one.
+                # References :
+                #1. Differential equations with applications
+                # and historical notes - George E. Simmons
+                #2. https://math.okstate.edu/people/binegar/2233-S99/2233-l12.pdf
+                if numerator:
+                    # If (dP/dy - dQ/dx) / Q = f(x)
+                    # then exp(integral(f(x))*equation becomes exact
+                    factor = simplify(numerator/n)
+                    variables = factor.free_symbols
+                    if len(variables) == 1 and x == variables.pop():
+                        factor = exp(Integral(factor).doit())
+                        m*= factor
+                        n*= factor
+                        self._wilds_match[P] = m.subs(y,fx)
```

Comment:
There should be spaces after commas and on both sides of something like `*=`. See PEP 8.